RevenueCat/Scripts/VirtualCurrencies.cs

```diff
@@ -0,0 +1,31 @@
+using System.Collections.Generic;
+using RevenueCat.SimpleJSON;
+using static RevenueCat.Utilities;
+
+public partial class Purchases
+{
+    /// <summary>
+    /// The VirtualCurrencies object contains all the virtual currencies associated to the user.
+    /// </summary>
+    public class VirtualCurrencies
+    {
+        /// <summary>
+        /// Map of all VirtualCurrency objects keyed by virtual currency code.
+        /// </summary>
+        public readonly Dictionary<string, VirtualCurrency> All;
```

Comment:
Should we use a `ReadOnlyDictionary` as the public API? If I understand correctly, this would allow developers to modify the dictionary after they obtain it and we might want to avoid that.

I think we haven't used that before... But feels like we should 😅